### PR TITLE
ICDS Dashboard | Monthly Register performace

### PR DESCRIPTION
**Pull request by lwyszomi**
Hi @emord,

I added some improvements to the Monthly Register PDF export.

Please let me know if you have any questions.

Regards,
Łukasz

### CODE REVIEW

File: custom/icds_reports/tests/reports/test_issnip_monthly_register.py [hunk 1]
```diff
@@ -93,11 +105,13 @@ def test_child_health_monthly_ucr(self):
 
     def test_agg_child_health_monthly(self):
         config = {
-            'awc_id': 'a3',
+            'awc_id': ['a3'],
```
Comment (emord) [inline custom/icds_reports/tests/reports/test_issnip_monthly_register.py]: Can you add a test that checks multiple AWCs?

**Review by emord — COMMENTED** [review]
Looks good. just need to add a test

File: custom/icds_reports/tests/reports/test_issnip_monthly_register.py (hunk 1 above)
Comment (lwyszomi) [reply custom/icds_reports/tests/reports/test_issnip_monthly_register.py]: @emord I added tests, Sorry for the delay.

File: custom/icds_reports/reports/issnip_monthly_register.py [hunk 2]
```diff
@@ -94,32 +95,32 @@ def css_record_monthly(self):
...
             )
         )
-        return data[0] if data else None
+        return data
 
     @cached_property
     def infrastructure_data(self):
-        data = AWCInfrastructureUCR(self.config).data
-        return list(data.values())[-1] if data else None
+        data = AWCInfrastructureUCR(self.config.copy()).data
```
Comment (emord) [inline custom/icds_reports/reports/issnip_monthly_register.py]: Why did you add the `.copy()` here? can these classes modify the config?

File: custom/icds_reports/reports/issnip_monthly_register.py [hunk 3]
```diff
@@ -287,4 +288,32 @@ def agg_child_health_monthly(self):
...
-        return data[0] if data else None
+        return data
+
+    def get_awc_name(self, awc_id):
+        return SQLLocation.objects.get(location_id=awc_id).name,
+
+    @cached_property
+    def to_pdf_format(self):
+        for awc in self.config['awc_id']:
+            agg_awc_monthly_data = [x for x in self.agg_awc_monthly_data if x['awc_id'] == awc]
```
Comment (emord) [inline custom/icds_reports/reports/issnip_monthly_register.py]: This would be much nicer/faster if you created a dictionary `{awc_id: data}` for each of these

File: custom/icds_reports/tests/reports/test_issnip_monthly_register.py [hunk 4]
```diff
@@ -130,3 +145,215 @@ def test_agg_child_health_monthly(self):
...
+
+    def test_agg_awc_monthly_data_multiple_locations(self):
+        config = {
+            'awc_id': ['a48', 'a3'],
+            'month': datetime(2017, 5, 1).date(),
+            'domain': 'icds-cas'
+        }
+        with mock.patch('custom.icds_reports.reports.issnip_monthly_register.ISSNIPMonthlyReport.get_awc_name',
+                        return_value='a48'):
+            data_a48 = list(ISSNIPMonthlyReport(config=config).to_pdf_format)[0]['agg_awc_monthly_data']
```
Comment (emord) [inline custom/icds_reports/tests/reports/test_issnip_monthly_register.py]: Would be nicer to only call this once.

**Review by emord — APPROVED** [review]
Can you follow up on my comments in a separate PR?

File: custom/icds_reports/reports/issnip_monthly_register.py (hunk 2 above)
Comment (lwyszomi) [reply custom/icds_reports/reports/issnip_monthly_register.py]: @emord Because ```clean_IN_filter_value(self.config, 'awc_id')``` function modify the config.

Comment (emord) [reply custom/icds_reports/reports/issnip_monthly_register.py]: Couldn't you do that once and pass in the cleaned version?
